### PR TITLE
onnxruntime v1.10.0

### PR DESCRIPTION
**Pull request by garymm**
* Remove osx_64_numpy1.18python3.6 since v1.10.0 dropped support for
  Python 3.6 on MacOS as noted in
  <https://github.com/microsoft/onnxruntime/releases/tag/v1.10.0>.
* Remove patch since it's no longer needed

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

File: .ci_support/osx_64_numpy1.18python3.7.____cpython.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 MACOSX_DEPLOYMENT_TARGET:
-- '10.14'
+- '10.9'
```
Comment (garymm) [inline .ci_support/osx_64_numpy1.18python3.7.____cpython.yaml:2]: @xhochy I think this leads to failures in the MacOS builds. Not sure the best way to fix it.

Comment (xhochy) [reply .ci_support/osx_64_numpy1.18python3.7.____cpython.yaml:2]: This needs to go into `recipe/conda_build_config.yaml`. All the `.ci_support` files are auto-generated.

Comment (xhochy) [reply .ci_support/osx_64_numpy1.18python3.7.____cpython.yaml:2]: Shouldn't 10.12 be sufficent?

Comment (garymm) [reply .ci_support/osx_64_numpy1.18python3.7.____cpython.yaml]: Added to recipe/conda_build_config.yaml.

I don't remember the details but when I updated to onnxruntime 1.9.0, I think MacOS 10.14 was the lowest version that worked. I think lower versions had some missing C++ standard library support.

**Comment by xhochy** [comment]
Thanks @garymm !

**Pull request merged**